PseudoGantry/lib/PseudoGantry/PseudoAxis.cpp

```diff
@@ -0,0 +1,121 @@
+#include "PseudoAxis.h"
+
+void set_up_encoder(PseudoEncoder *encoder, void (*isr_motor_pulse)(void))
+{
+    pinMode(encoder->motor_pulse_pin, INPUT_PULLUP);
+    pinMode(encoder->channel_a_out, OUTPUT);
+    digitalWrite(encoder->channel_a_out, LOW);
+    attachInterrupt(digitalPinToInterrupt(encoder->motor_pulse_pin), isr_motor_pulse, CHANGE);
+}
+
+bool toggle_encoder_output(PseudoAxis *pseudo_axis)
+{
+    pseudo_axis->skip_counter++;
+    // skip steps in the beginning
+    if (pseudo_axis->skip_counter <= pseudo_axis->changes_to_skip) return false;
+    // else start toggling until the next set of STEPS
+    else if ((pseudo_axis->skip_counter <= (2 * pseudo_axis->steps_for_ratio)) || (pseudo_axis->changes_to_skip == 0))
+    {   
+        bool motor_pin_status = digitalRead(pseudo_axis->encoder.motor_pulse_pin);
+        // Serial.print("motor pin:    "); Serial.println(motor_pin_status);
+        digitalWrite((pseudo_axis->encoder).channel_a_out, motor_pin_status ^ HIGH);
+        // check if encoder count needs to be incremented
+        // if HIGH it means triggered on rising
+        if (motor_pin_status) return true;
+        else return false;
+    }
+    else
+    {
+        // this is assigned 1 and not 0 because at this point our step_count_x is ((2 * STEPS) + 1)
+        // which means we are onto the next set of 8, and 17 would be the beginnig of that set
+        pseudo_axis->skip_counter = 1;
+        return false;
+    }
+}
+
+void isr_motor_pulse(PseudoAxis *pseudo_axis)
+{   
+    if (!toggle_encoder_output(pseudo_axis)) return;
+
+    bool direction = digitalRead(pseudo_axis->motor_dir_pin);
+    // Serial.print("direction:    "); Serial.println(direction);
+    // Serial.print("to skip:      "); Serial.println(pseudo_axis->changes_to_skip);
+
+    if (direction)
+    {
+        if (pseudo_axis->motor_position_current < pseudo_axis->axis_length_counts)
+        {
+            pseudo_axis->motor_position_current++;
+            // check if limit switch was pressed before this move
+            // checks whether the homing routine is being run
+            if (!pseudo_axis->ls_home.status)
+            { // if !PRESSED where PRESSED = 0
```

Comment:
Would be clearer to remove the comment and write `if (pseudo_axis->ls_home.status == PRESSED)`